extractor/filesystem/containers/k8simage/k8simage_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8simage_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem/containers/k8simage"
+	"github.com/google/osv-scalibr/extractor/filesystem/simplefileapi"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scalibr/testing/extracttest"
+)
+
+func TestFileRequired(t *testing.T) {
+	extr := k8simage.New(k8simage.DefaultConfig())
+
+	tests := []struct {
+		name string
+		path string
+		want bool
+	}{
+		{
+			name: "K8s Deployment",
+			path: "testdata/deployment.yaml",
+			want: true,
+		},
+		{
+			name: "K8s ReplicaSet",
+			path: "testdata/frontend.yaml",
+			want: true,
+		},
+		{
+			name: "K8s Job",
+			path: "testdata/job.yaml",
+			want: true,
+		},
+		{
+			name: "K8s Pod",
+			path: "testdata/pod.yaml",
+			want: true,
+		},
+		{
+			name: "K8s Service",
+			path: "testdata/service.yaml",
+			want: true,
+		},
+		{
+			name: "K8s StatefulSet",
+			path: "testdata/StatefulSet.yaml",
+			want: true,
+		},
+		{
+			name: "K8s multi config",
+			path: "testdata/web.yaml",
+			want: true,
+		},
+		{
+			name: "K8s missing apiVersion field",
+			path: "testdata/not-valid-k8s.yaml",
+			want: false,
+		},
+		{
+			name: "K8s missing kind field",
+			path: "testdata/not-valid-k8s-2.yaml",
+			want: false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			isRequired := extr.FileRequired(simplefileapi.New(tc.path, nil))
+			if isRequired != tc.want {
+				t.Fatalf("FileRequired(%s): got %v, want %v", tc.path, isRequired, tc.want)
+			}
+		})
+	}
+}
+
+func TestExtract(t *testing.T) {
+	tests := []struct {
+		name         string
+		path         string
+		cfg          k8simage.Config
+		wantPackages []*extractor.Package
+	}{
+		{
```

Comment:
how about a testcase that checks for the default behavior if not tag is set in the image property? As per your code and the spec this should return "latest" as tag.